bubble.rb

```diff
@@ -0,0 +1,58 @@
+def bubble_sort(array)
+  i = 0
+  j = 0
+  not_sorted = true
+  while i < array.length - 1 && not_sorted
+    not_sorted = false
+    j = 0
+    while j < array.length - 1
+      if array[j].to_i > array[j + 1].to_i
+        temp = array[j]
+        array[j] = array[j + 1]
+        array[j + 1] = temp
+        not_sorted = true
+      end
+      j += 1
+    end
+    i += 1
+  end
+end
+
+def bubble_sort_by(array)
+  i = 0
+  j = 0
+
+  not_sorted = true
+  while i < array.length - 1 && not_sorted
+    not_sorted = false
+    j = 0
+    while j < array.length - 1
+      comparison = yield(array[j], array[j + 1])
+      if comparison.positive?
+        temp = array[j]
+        array[j] = array[j + 1]
+        array[j + 1] = temp
+        not_sorted = true
+      end
+      j += 1
+    end
+    i += 1
+  end
+  print array
```

Comment:
let us return an array instead of printing array, note method ought to return not print